packages/server/lib/models/data.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:enum_to_string/enum_to_string.dart';
+
+abstract class Data {
+  late DataType dataType;
+
+  Data(this.dataType);
+
+  Data.fromJson(Map<String, dynamic> json) {
+    var type = EnumToString.fromString(DataType.values, json['type']);
+    if (type == null)
```

Comment:
We have great operator for that dataType = type ?? DataType.Unknown; 
It does exactly what you wrote